phosphor-power-supply/power_supply.cpp

```diff
@@ -1164,8 +1164,9 @@ void PowerSupply::getInputVoltage(double& actualInputVoltage,
 void PowerSupply::checkAvailability()
 {
     bool origAvailability = available;
-    available = present && !hasInputFault() && !hasVINUVFault() &&
-                !hasPSKillFault() && !hasIoutOCFault();
+    bool faulted = isPowerOn(bus) &&
```

Comment:
This function runs every on every power supply, so it isn't optimal to do a new D-Bus call per second per PS here.

The manager class is already tracking the power state, so I think it would be best for the PSU class to get it from there.  There's a few ways to do that, none is jumping out at me as being the best:

1. Pass a reference to the manager class into the constructor
2. Have manager's powerStateChanged function also call into the PSUs to tell them, and also maybe pass in current power state to constructor
3. pass a function into the constructor to get power state, and have it return manager->powerOn


Also, you can just do this upstream.  we should be able to review just as fast there.

Reply:
@spinler Can this PR be closed since a different fix with more review was merged upstream?

Reply:
yea.